client/my-sites/site-settings/form-general.jsx

```diff
@@ -357,18 +357,25 @@ export class SiteSettingsFormGeneral extends Component {
 			fields,
 			isRequestingSettings,
 			eventTracker,
+			selectedEditor,
 			siteIsJetpack,
 			siteIsAtomic,
 			translate,
 		} = this.props;
 		const blogPublic = parseInt( fields.blog_public, 10 );
 		const wpcomComingSoon = parseInt( fields.wpcom_coming_soon, 10 );
 
-		const isNonAtomicJetpackSite = siteIsJetpack && ! siteIsAtomic;
+		const siteIsSimple = ! siteIsJetpack;
+		const siteIsGutenbergAtomic = siteIsAtomic && selectedEditor && selectedEditor !== 'classic';
+		const siteIsClassicAtomic = siteIsAtomic && selectedEditor === 'classic';
+		const siteIsAlreadyPrivate = blogPublic === -1;
+
+		const mayGoPrivate =
+			siteIsSimple || siteIsGutenbergAtomic || ( siteIsClassicAtomic && siteIsAlreadyPrivate );
```

Comment:
Won't this result in a case like this:
* a site is private, Atomic, & Gutenberg
* Classic editor is selected
* Customer sets site to private
* The option disappears without warning

Reply:
Is it possible to go back to classic editor? If that's the case, then we should add a warning there too - saying you can't do it with a private site.

Reply:
Only HEs may move the user back to classic editor, I think the best we can do is add a warning on user RC ( C02DQP0FP-slack-1584093599198300 )